commonItems/Log.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
```

Comment:
Is this a case where you want to replicate the existing logger? It looks like C# has some default tooling: https://docs.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line
And ways to write one that'll work the way you want: https://docs.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider

Reply:
This seems overly complicated. What would the benefit of using ILogger be?

Reply:
Just a nudge towards considering the standard tools when available (when starting over like this, its worth abandoning the existing converter approaches if something better can be found). And I'd guess they should be more efficient. That said, this article suggests some other options would be better if using a standard solution: https://michaelscodingspot.com/logging-in-dotnet/

Reply:
Out of the 3rd party libraries I'd probably choose log4net, but they don't have a Progress log level, so it's not directly usable.